File: app/src/main/java/ua/te/hackathon/smartcity2015/api/model/DateTime.java
package ua.te.hackathon.smartcity2015.api.model;

import org.joda.time.format.DateTimeFormat;
import org.joda.time.format.DateTimeFormatter;

import java.util.Calendar;
import java.util.Formatter;

/**
 * Created by nk91 on 14.02.16.
 */
public class DateTime {
    private static final String FORMAT = "HH:mm DD-MM-YY";
    public static final DateTimeFormatter DATETIME_FORMATTER = DateTimeFormat.forPattern(FORMAT);
    private static final String DATE_FORMAT = "DD-MM-YY";
    public static final DateTimeFormatter DATE_FORMATTER = DateTimeFormat.forPattern(DATE_FORMAT);
    private static final String TIME_FORMAT = "HH:mm";
    public static final DateTimeFormatter TIME_FORMATTER = DateTimeFormat.forPattern(TIME_FORMAT);
    private int years;
    private int months;
    private int days;
    private int hours;
    private int minutes;
    private int seconds;

    public void setYears(int years) {
        this.years = years;
    }

    public void setMonths(int months) {
        this.months = months;
    }

    public void setDays(int days) {
        this.days = days;
    }

    public void setHours(int hours) {
        this.hours = hours;
    }

    public void setMinutes(int minutes) {
        this.minutes = minutes;
    }

    public void setSeconds(int seconds) {
        this.seconds = seconds;
    }

    public String get() {
        org.joda.time.DateTime dateTime = new org.joda.time.DateTime(years, months, days, hours, minutes, seconds);
        return DATETIME_FORMATTER.print(dateTime);
    }

    public String getTime() {
        org.joda.time.DateTime dateTime = new org.joda.time.DateTime(years, months, days, hours, minutes, seconds);
        return TIME_FORMATTER.print(dateTime);
    }

    public String getDate() {
        org.joda.time.DateTime dateTime = new org.joda.time.DateTime(years, months, days, hours, minutes, seconds);
        return DATE_FORMATTER.print(dateTime);
    }

    public long getMillis() {
        org.joda.time.DateTime dateTime = new org.joda.time.DateTime(years, months, days, hours, minutes, seconds);
        return dateTime.getMillis();
    }
}

File: app/src/main/java/ua/te/hackathon/smartcity2015/api/model/DateTime.kt
package ua.te.hackathon.smartcity2015.api.model

import org.joda.time.format.DateTimeFormat

/**
 * Created by nk91 on 14.02.16.
 */
class DateTime {
  private var years: Int = 0
  private var months: Int = 0
  private var days: Int = 0
  private var hours: Int = 0
  private var minutes: Int = 0
  private var seconds: Int = 0

  fun setYears(years: Int) {
    this.years = years
  }

  fun setMonths(months: Int) {
    this.months = months
  }

  fun setDays(days: Int) {
    this.days = days
  }

  fun setHours(hours: Int) {
    this.hours = hours
  }

  fun setMinutes(minutes: Int) {
    this.minutes = minutes
  }

  fun setSeconds(seconds: Int) {
    this.seconds = seconds
  }

  fun get(): String {
    val dateTime = org.joda.time.DateTime(years, months, days, hours, minutes, seconds)
    return DATETIME_FORMATTER.print(dateTime)
  }

  val time: String
    get() {
      val dateTime = org.joda.time.DateTime(years, months, days, hours, minutes, seconds)
      return TIME_FORMATTER.print(dateTime)
    }

  val date: String
    get() {
      val dateTime = org.joda.time.DateTime(years, months, days, hours, minutes, seconds)
      return DATE_FORMATTER.print(dateTime)
    }

  val millis: Long
    get() {
      val dateTime = org.joda.time.DateTime(years, months, days, hours, minutes, seconds)
      return dateTime.millis
    }

  companion object {
    private val FORMAT = "HH:mm DD-MM-YY"
    val DATETIME_FORMATTER = DateTimeFormat.forPattern(FORMAT)
    private val DATE_FORMAT = "DD-MM-YY"
    val DATE_FORMATTER = DateTimeFormat.forPattern(DATE_FORMAT)
    private val TIME_FORMAT = "HH:mm"
    val TIME_FORMATTER = DateTimeFormat.forPattern(TIME_FORMAT)
  }
}
